eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -39,9 +39,15 @@ steps:
               Write-Host "Repository $($repository.Name) has already been initialized. Skipping this step."
             } else {
               Write-Host "Repository $($repository.Name) is being initialized."
+
+              Write-Host "git clone --no-checkout --filter=tree:0 git://github.com/$($repository.Name) ."
               git clone --no-checkout --filter=tree:0 git://github.com/$($repository.Name) .
+
+              Write-Host "git sparse-checkout init"
               git sparse-checkout init
-              git sparse-checkout set eng
+
+              Write-Host "git sparse-checkout add /eng"
+              git sparse-checkout add /eng
```

Comment:
I worry that relying on the implicit sparse checkout default makes the behavior less obvious for people who haven't inspected the contents of `.git/info/sparse-checkout`, and also vulnerable to future changes (however unlikely) in `git sparse-checkout init` behavior.

```suggestion
              git sparse-checkout add '/*' '!/*/' '/eng'
```

Reply:
I started with this but after reading the docs I thought it would be better to just use init and add. However if we want to be explicit here then we should do a set with those paths.

Reply:
Maybe just adding a comment above the `add` explaining about the default values cloning the root directory would be enough context?

Reply:
I switch to explicitly including them.